pkg/controller/admissionchecks/multikueue/batchjob_adapter.go

```diff
@@ -30,14 +32,41 @@ type batchJobAdapter struct{}
 
 var _ jobAdapter = (*batchJobAdapter)(nil)
 
-func (b *batchJobAdapter) CreateRemoteObject(ctx context.Context, localClient client.Client, remoteClient client.Client, key types.NamespacedName, workloadName string) error {
+func (b *batchJobAdapter) SyncJob(ctx context.Context, localClient client.Client, remoteClient client.Client, key types.NamespacedName, workloadName, origin string) error {
 	localJob := batchv1.Job{}
 	err := localClient.Get(ctx, key, &localJob)
 	if err != nil {
 		return err
 	}
 
-	remoteJob := batchv1.Job{
+	remoteJob := batchv1.Job{}
+	err = remoteClient.Get(ctx, key, &remoteJob)
+	if client.IgnoreNotFound(err) != nil {
+		return err
+	}
+
+	// the remote job exists
+	if err == nil {
+		// This will no longer be necessary when batchJob will support live status update, by then
+		// we should only sync the Status of the job if it's "Finished".
+		remoteFinished := false
+		for _, c := range remoteJob.Status.Conditions {
```

Comment:
nit: Please use utility function (IsStatusConditionTrue) instead of looping here

Reply:
these are not metav1.Conditions

Reply:
sg